drud/ddev
# Generate nginx/apache configs on host and mount into container, fixes #2263, fixes #2084, fixes #1889, fixes #1957

## The Problem/Issue/Bug:

The nginx configuration for ddev has long been much too complicated and hard to maintain.

See #2263 

- [x]  Move the generation of the files to the host
- [x] Mount the generated files when starting the web container
- [x] Use a single stanza for both port 80 and port 443
- [ ]  Release notes - snippets are deprecated, still supported, but migrate them into full conf
- [ ]  Release notes: Migrating nginx-site.conf from .ddev to .ddev/nginx_full (remember to change root)
- [x] On upgrade, move an existing nginx-site.conf into nginx_full? Warn them while doing it? Change root?
- [x]  Add README files into .ddev/nginx, .ddev/nginx_full, and apache directories
- [x]  Make sure MULTIPLE DOCROOTS is in docs
- [x] Add second docroot example into nginx_full
- [x] Add gitignores

## How this PR Solves The Problem:

## Manual Testing Instructions:

* Follow the instructions in .ddev/nginx_full/nginx-site.conf to change the config (and remove #ddev-generated)
* `ddev start` and check for appropriate nginx behavior
* Follow the example file in nginx_full to add a second docroot. Test
* Same with apache

## Automated Testing Overview:


## Related Issue Link(s):

## Release/Deployment notes:
